pkg/crc/machine/machine.go

```diff
@@ -171,13 +171,22 @@ func Start(startConfig StartConfig) (StartResult, error) {
 		return *result, err
 	}
 
-	hostIP, err := network.DetermineHostIP(instanceIP)
+	var hostIP string
+	for i := 0; i < 30; i++ {
```

Comment:
This complete logic can move to network package only.

```patch
diff --git a/pkg/crc/machine/machine.go b/pkg/crc/machine/machine.go
index a0b3403..a995b07 100644
--- a/pkg/crc/machine/machine.go
+++ b/pkg/crc/machine/machine.go
@@ -3,6 +3,7 @@ package machine
 import (
 	"encoding/json"
 	"fmt"
+	"github.com/code-ready/crc/pkg/crc/network"
 	"io/ioutil"
 	"path/filepath"
 	"time"
@@ -11,7 +12,6 @@ import (
 	"github.com/code-ready/crc/pkg/crc/errors"
 	"github.com/code-ready/crc/pkg/crc/logging"
 
-	"github.com/code-ready/crc/pkg/crc/network"
 	"github.com/code-ready/crc/pkg/crc/systemd"
 	crcos "github.com/code-ready/crc/pkg/os"
 
@@ -171,17 +171,7 @@ func Start(startConfig StartConfig) (StartResult, error) {
 		return *result, err
 	}
 
-	var hostIP string
-	for i := 0; i < 30; i++ {
-		hostIP, err = network.DetermineHostIP(instanceIP)
-		if err == nil {
-			logging.DebugF("Bridge IP on the host: %s", hostIP)
-			break
-		} else {
-			logging.DebugF("Error finding host IP (%v) - retrying", err)
-		}
-		time.Sleep(2 * time.Second)
-	}
+	hostIP, err := network.DetermineHostIPWithRetry(instanceIP, 30)
 	if err != nil {
 		logging.ErrorF("Error determining host IP: %v", err)
 		result.Error = err.Error()
diff --git a/pkg/crc/network/utils.go b/pkg/crc/network/utils.go
index abf8725..d4cb88c 100644
--- a/pkg/crc/network/utils.go
+++ b/pkg/crc/network/utils.go
@@ -3,6 +3,7 @@ package network
 import (
 	"fmt"
 	"net"
+	"time"
 
 	"github.com/code-ready/crc/pkg/crc/errors"
 
@@ -81,3 +82,15 @@ func DetermineHostIP(instanceIP string) (string, error) {
 
 	return "", errors.New("unknown error occurred")
 }
+
+
+func DetermineHostIPWithRetry(instanceIP string, retry int) (string, error) {
+	for i := 0; i < retry; i++ {
+		hostIP, err := DetermineHostIP(instanceIP)
+		if err == nil {
+			return hostIP, nil
+		}
+		time.Sleep(2 * time.Second)
+	}
+	return "", fmt.Errorf("Not able to find host IP after %d retry", retry)
+}
```

Reply:
agree